tasks/install_fpga_packages.yml

```diff
@@ -0,0 +1,143 @@
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Fail if fpga_rsu_driver_pkg_url is undefined and empty
+  fail:
+    msg: "fpga_rsu_driver_pkg_url is a mandatory variable"
+  when:
+    - not (( fpga_rsu_driver_pkg_url is defined) and (fpga_rsu_driver_pkg_url | length > 0))
+
+- name: Check and create FPGA driver directory if not exist
+  block:
+    - name: check fpga driver package directory exist
+      stat:
+        path: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_dir_result
+
+    - name: Create fpga driver package directory if not exist
+      file:
+        path: "{{ fpga_rsu_driver_dir }}"
+        state: directory
+        mode: 0755
+      when: not (fpga_driver_dir_result.stat.exists | bool)
+
+- name: Download and untar FPGA driver tarball
+  block:
+    - name: Download fpga driver tarball
+      get_url:
+        url: "{{ fpga_rsu_driver_pkg_url }}"
+        dest: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+      register: fpga_driver_tarball_downloaded
+      failed_when: false
+
+    - name: Fail if fpga driver tarball download fail.
+      fail:
+        msg: "FPGA Driver tarball download failed. Please check correct tarball URL is provided."
+      when:
+        - fpga_driver_tarball_downloaded is failed
+
+    - name: untar fpga driver tarball
+      unarchive:
+        src: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+        dest: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_tarball_untar
+      failed_when: false
+
+- name: Setting fact and get rpm list
+  block:
+    - name: setting fact for fpga driver rpm directory
+      set_fact:
+        fpga_driver_rpm_dir: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name | splitext | first | splitext | first }}"
+
+    - name: get rpm list for installation
+      shell: ls -1 *.rpm
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_list_of_rpms
+      when: fpga_driver_tarball_untar is not failed
+
+    - name: get version of opae package
+      shell: rpm -qp --queryformat '%{VERSION}-%{RELEASE}' "opae-libs*"
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_package_version
+      when: opae_list_of_rpms.stdout_lines | length > 0
+
+- name: Install packages and verify installation
+  block:
+    - name: install fpga drivers
+      shell:
+        rpm -ivh {{ opae_list_of_rpms.stdout_lines | join(' ') }}
```

Comment:
What will happen if rpms are already installed? Should we update rpms in this step (use `-U` flag)?